helm/distributed-idol/templates/dih/statefulset.yaml

```diff
@@ -34,116 +26,32 @@ spec:
         - sh
         - "-c"
         - |
-{{ tpl (.Files.Get "resources/dih_init.sh") . | indent 10 }}
+{{ tpl ($root.Files.Get "resources/dih_init.sh") $root | indent 10 }}
         volumeMounts:
         - name: config-map
           mountPath: /mnt/config-map
         - name: dih-persistent-storage
           mountPath: /mnt/store/dih
       imagePullSecrets:
-        {{- range .Values.global.imagePullSecrets }}
+        {{- range $root.Values.global.imagePullSecrets }}
         - name: {{ . }}
         {{- end }}
-      containers:
-      - name: {{ $component.name }}
-        image: {{ include "idol-library.idolImage" (dict "root" . "idolImage" $component.idolImage )}}
-        imagePullPolicy: IfNotPresent
-        command: ["/bin/sh"]
-        args: 
-        - -c
-        - |
-          cd /dih
-          ln -sf /opt/idol/dih/data ./data
-          if [ -e /opt/idol/dih/data/dih.cfg ]
-          then
-            echo Using existing dih.cfg
-          else
-            cp /opt/idol/dih/data/dih.install.cfg /opt/idol/dih/data/dih.cfg
-          fi
-          ln -sf /opt/idol/dih/data/dih.cfg ./dih.cfg
-          ./run_idol.sh
-        
-        lifecycle:
-          preStop:
-            exec:
-              command:
-              - /bin/bash
-              - "-c"
-              - |
-{{ tpl (.Files.Get "resources/dih_preStop.sh") . | indent 16 }}
-        volumeMounts:
-        - name: dih-persistent-storage
-          mountPath: /opt/idol/dih/data
-        {{- range $component.additionalVolumeMounts }}
-        - {{ . | toYaml | nindent 10 }}
-        {{- end }}
-        env:
-        - name: IDOL_COMPONENT_CFG
-          # Override default in newer IDOL containers
-          value: "/dih/dih.cfg"
-        {{- if $component.envConfigMap }}
-        envFrom:
-        - configMapRef: {{ $component.envConfigMap | quote }}
-        {{ end }}
-        livenessProbe:
-          httpGet:
-            path: /a=getpid
-            port: {{ $component.aciPort | int }}
-{{- template "idol-library.standardLivenessProbe" $component.livenessProbe }}
-        ports:
-        - containerPort: {{ $component.aciPort | int }}
-          name: aci-port
-          protocol: TCP
-        - containerPort: {{ $component.servicePort | int }}
-          name: service-port
-          protocol: TCP
-        - containerPort: {{ $component.indexPort | int }}
-          name: index-port
-          protocol: TCP
-{{- /*
- Exporter monitors child fullness - unused in mirrored setup 
-*/}}
-{{- if and (.Values.autoscaling.enabled) (not .Values.setupMirrored) }}
-      - name: prometheus-exporter
-        image: python:3.11
-{{- if (coalesce .Values.global.http_proxy .Values.global.https_proxy) }}
-        envFrom:
-          - configMapRef:
-              name: http-proxy-config
+{{- if (dig "podSecurityContext" "enabled" false ($component | merge (dict))) }}
+      securityContext: {{- omit $component.podSecurityContext "enabled" | toYaml | nindent 8 }}
```

Comment:
You shouldn't need these here now that you are merging with `idol-library.aciserver.statefulset.base.v1`

Reply:
https://github.com/opentext-idol/idol-containers-toolkit/pull/80/commits/d6014e6b1935ce5dc204160fb2295feb6739b413 https://github.com/opentext-idol/idol-containers-toolkit/pull/80/commits/980443cf229f4301b79143f0eb401170152edc16